doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst

```diff
@@ -0,0 +1,336 @@
+
+Keys stored in the KMU can be used in standard PSA cryptographic operations for encryption, decryption, signing a hash or a message, and verifying a hash or a message, given that the corresponding ``PSA_KEY_USAGE_*`` flags are set.
+
+Keys with the usage schemes Protected (``CRACEN_KMU_KEY_USAGE_SCHEME_PROTECTED``) and Seed (``CRACEN_KMU_KEY_USAGE_SCHEME_SEED``) can push data from the slots directly to CRACEN registers that are not accessible by the CPU.
+These usage schemes are supported for pushing symmetric keys used for cipher operations, and for seeds used by the CRACEN IKG.
+
+Keys with the usage scheme Raw (``CRACEN_KMU_KEY_USAGE_SCHEME_RAW``), such as symmetric keys, are temporarily pushed to a RAM location by the CRACEN driver, and then loaded by CRACEN into the asymmetric engine before running operations like sign or verify.
+
+Key slots with the usage scheme Encrypted (``CRACEN_KMU_KEY_USAGE_SCHEME_ENCRYPTED``) also have to be decrypted to a temporary push location in RAM before they are used by CRACEN, which is handled by the CRACEN driver.
+
+When the application is built with TF-M, this temporary push location is protected inside the secure processing environment to avoid exposing the key material to the non-secure application.
+If TF-M is not used, the keys are pushed to a reserved RAM area at location 0x20000000-0x20000064 (``kmu_push_area``).
+
+Some of the KMU-specific error codes you might encounter when using the KMU keys are:
+
+* PSA_ERROR_INVALID_HANDLE: Attempting an operation on an empty KMU slot.
+* PSA_ERROR_NOT_PERMITTED: Attempting an operation on a revoked key.
+* PSA_ERROR_HARDWARE_FAILURE: The key slot has invalid data.
+* PSA_ERROR_CORRUPTION_DETECTED: The key slot has invalid data.
+
+Configuration
+*************
+
+The CRACEN peripheral does not require any configuration.
+Its operation is ensured by hardware.
+
+To enable KMU support, set the following Kconfig options:
+
+.. code-block:: console
+
+   CONFIG_NRF_SECURITY=y
+   CONFIG_MBEDTLS_PSA_CRYPTO_C=y
+   CONFIG_PSA_CRYPTO_DRIVER_CRACEN=y
+
+As an example, enabling support for the Encrypted key usage scheme (``CRACEN_KMU_KEY_USAGE_SCHEME_ENCRYPTED``) should look as follows:
+
+.. code-block:: console
+
+   CONFIG_PSA_WANT_KEY_TYPE_AES=y
+   CONFIG_PSA_WANT_AES_KEY_SIZE_256=y
+   CONFIG_PSA_WANT_ALG_ECB_NO_PADDING=y
+   CONFIG_PSA_WANT_ALG_CMAC=y
+   CONFIG_PSA_WANT_ALG_SP800_108_COUNTER_CMAC=y
+   CONFIG_PSA_WANT_ALG_GCM=y
+
+The configuration is enabling the key type (AES) and the key size (256 bits) supported by the Encrypted usage scheme as explained in the :ref:`ug_nrf54l_crypto_kmu_supported_key_types` section, and in addition enabling the following :ref:`cryptographic features <nrf_security_driver_config>` supported by the CRACEN driver:
+
+* Cipher mode: AES ECB (Electronic CodeBook) mode, no padding
+* Message Authentication Code (MAC) cipher: cipher-based MAC (CMAC) cipher
+* Key derivation function (KDF) support: SP800-108 CMAC in counter mode
```

Comment:
Will add CONFIG_PSA_WANT_ALG_SP800_108_COUNTER_CMAC in the KDF section in the nRF Security lib (https://docs.nordicsemi.com/bundle/ncs-2.7.99-cs1/page/nrf/libraries/security/nrf_security/doc/driver_config.html#key_derivation_function_configurations).

Reply:
Kconfig added in PR https://github.com/nrfconnect/sdk-nrf/pull/17804, a draft PR with work in progress.